### PR TITLE
update to 3.0

### PR DESCRIPTION
**Pull request by bosr**
Hi,
- Dependencies updates
- slight change for `UDPSocket.receive(upTo:timingOut:)` to be consistent with TCP

[edit: updated to 3.0 in the last commit]

### CODE REVIEW

**Comment by paulofaria** [comment]
@bosr yoooo! thanks a lot for this. This updates to the latest snapshot right? This opens some questions. It's not guaranteed that the last updates on the latests snapshots are going to land in Swift 3. So I think maybe it's best we use _swift-3.0-GM-CANDIDATE_ in our `.swift-version` files. I hope this change doesn't affect any code. 

And also. I'm done updating most of the modules on this monolith we use for development:

https://github.com/Zewo/Flux

This package contains the most important modules, bundled together in one module. The idea is to split this monolithic module into the proper modules that will reside in their own repos. This is what I'm about to do now. So we should sync this work so that we don't step on each other's toes. Luckily UDP is not a part of Zewo/Flux. So no harm done for now. You can add UDP there or if you want I can do it.

**Comment by bosr** [comment]
Thanks @paulofaria ! I'll do this asap for this repo (just to be clean). I saw your message on slack, but I had not understood you were already on it ! 👍

Regarding Zewo/Flux, I'll try to add UDP there, sure!

Regards

**Comment by bosr** [comment]
Hi @paulofaria!, here is my "plan" :) do you agree?
- I will push here the updates that were made during the review on UDP in Zewo/Flux.
- When merged here, I'll PR the git subrepo in Zewo/Zewo.

**Comment by bosr** [comment]
thanks, I added it for no reason

File: Package.swift [hunk 1]
```diff
@@ -3,6 +3,11 @@ import PackageDescription
 let package = Package(
     name: "UDP",
     dependencies: [
-        .Package(url: "https://github.com/VeniceX/IP.git", majorVersion: 0, minor: 12),
+        .Package(url: "https://github.com/Zewo/POSIX.git", majorVersion: 0, minor: 13),
+        .Package(url: "https://github.com/Zewo/IP.git", majorVersion: 0, minor: 13),
+
+
+        // test-only dependencies (not yet available in swiftpm: SE-0019)
```
Comment (paulofaria) [inline Package.swift:10]: great! (:

**Comment by bosr** [comment]
@paulofaria ,

Could you tell me what changed since you had me remove the extension on the testcases? :)

**Comment by paulofaria** [comment]
Haha! Sorry I wasn't clear. I didn't mean remove the extension. I mean remove the leading space that existed in:

``` swift
var allTests : [(String, (UDPSocketTests) -> () throws -> Void)]
             ^
```

to

``` swift
var allTests: [(String, (UDPSocketTests) -> () throws -> Void)]
            ^
```

That typo was everywhere since we copy-pasted the test template. I'm struggling to get rid of those until today haha.

**Comment by bosr** [comment]
Ha ha, dumb me! Sorry I did not get this the 1st time. Cheers!

**Comment by paulofaria** [comment]
Just waiting for travis and I think we're good to merge. OK to you?

**Comment by bosr** [comment]
yup

**Comment by bosr** [comment]
damn, you're fast, I was doing it :)

**Comment by paulofaria** [comment]
I did everything through GitHub web, which is very handy, but also can lead to typos like that one, haha.

**Comment by paulofaria** [comment]
Everything's good. Wanna do the honors? Just make sure to use Squash and merge. (:

**Pull request merged**